wordpress-mobile/WordPress-Android
# Convert notes adapter to kotlin

### Description

This PR refactors the NotesAdapter class to Kotlin. This is intended to improve the further development of the features we are adding in the notifications refresh project.

-----

## To Test:

* Run the app using this branch (either build in Android Studio, or via one of the apks)
* Login to the app using an account which has some activity (with various kinds of notifications)
* Smoke test the notifications behavior:
  * Scroll the notifications list and ensure they all look correct
  * Tap on some of the notifications to ensure that the behavior is still correct (e.g. tap a like notification to see the list of users who liked the comment or post, etc.)

-----

## Regression Notes

1. Potential unintended areas of impact

Notifications

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual tests

3. What automated tests I added (or what prevented me from doing so)

With the code in its current state, adding automated tests would exceed the scope allotted for this work.

-----

## PR Submission Checklist:


-----

## Testing Checklist:
